packages/ui/src/components/editor/input/Euler/index.tsx

```diff
@@ -43,66 +43,71 @@ type EulerInputProps = {
   unit?: string
 }
 
+const temp_Euler = new Euler()
+const temp_Quat = new Quaternion()
+
 /**
  * FileIEulerInputnput used to show EulerInput.
  *
  * @type {Object}
  */
 export const EulerInput = (props: EulerInputProps) => {
   const quaternion = useHookstate(props.quaternion)
-  const euler = useHookstate(new Euler().setFromQuaternion(quaternion.value, 'YXZ'))
+  const euler = useHookstate(temp_Euler.setFromQuaternion(props.quaternion, 'YXZ'))
 
   useEffect(() => {
-    euler.value.setFromQuaternion(quaternion.value, 'YXZ')
+    euler.set(temp_Euler.setFromQuaternion(quaternion.value, 'YXZ'))
   }, [props.quaternion])
 
   const onSetEuler = useCallback(
```

Comment:
we can remove the `useCallback` here since it will be memoized by react compiler

Reply:
done